src/main/java/com/limechain/network/protocol/grandpa/GrandpaEngine.java

```diff
@@ -71,45 +80,48 @@ private void handleResponderStreamMessage(byte[] message, GrandpaMessageType mes
         }
     }
 
-    private GrandpaMessageType getGrandpaMessageType(byte[] msg) {
-        if (msg.length == HANDSHAKE_LENGTH) {
+    private GrandpaMessageType getGrandpaMessageType(byte[] message) {
+        if (message.length == HANDSHAKE_LENGTH) {
             return GrandpaMessageType.HANDSHAKE;
         }
-        return GrandpaMessageType.getByType(msg[0]);
+        return GrandpaMessageType.getByType(message[0]);
     }
 
-    private void handleHandshake(byte[] msg, PeerId peerId, Stream stream) {
+    private void handleHandshake(byte[] message, PeerId peerId, Stream stream) {
         if (connectionManager.isGrandpaConnected(peerId)) {
             log.log(Level.INFO, "Received existing grandpa handshake from " + peerId);
         } else {
             connectionManager.addGrandpaStream(stream);
-            connectionManager.getPeerInfo(peerId).setNodeRole(msg[0]);
+            connectionManager.getPeerInfo(peerId).setNodeRole(message[0]);
             log.log(Level.INFO, "Received grandpa handshake from " + peerId);
         }
         writeHandshakeToStream(stream, peerId);
     }
 
-    private void handleNeighbourMessage(byte[] msg, PeerId peerId) {
-        ScaleCodecReader reader = new ScaleCodecReader(msg);
+    private void handleNeighbourMessage(byte[] message, PeerId peerId) {
+        ScaleCodecReader reader = new ScaleCodecReader(message);
         NeighbourMessage neighbourMessage = reader.read(new NeighbourMessageScaleReader());
         log.log(Level.INFO, "Received neighbour message from Peer " + peerId + "\n" + neighbourMessage);
-        if (syncedState.isWarpSyncFinished()
-                && neighbourMessage.getSetId().compareTo(syncedState.getSetId()) > 0) {
-            new Thread(() -> syncedState.updateSetData(neighbourMessage, peerId)).start();
-        }
+        new Thread(() -> syncedState.syncNeighbourMessage(neighbourMessage, peerId)).start();
     }
 
-    private void handleCommitMessage(byte[] msg, PeerId peerId) {
-        ScaleCodecReader reader = new ScaleCodecReader(msg);
+    private void handleCommitMessage(byte[] message, PeerId peerId) {
+        ScaleCodecReader reader = new ScaleCodecReader(message);
         CommitMessage commitMessage = reader.read(new CommitMessageScaleReader());
-        if (isBlockAlreadyReached(commitMessage.getVote().getBlockNumber())) {
-            log.log(Level.FINE, String.format("Received commit message for finalized block %d from peer %s",
-                            commitMessage.getVote().getBlockNumber(), peerId));
-            return;
+        synchronized (this) {
```

Comment:
Can we add a comment about the race condition? In the future someone might wonder why this part is `synchronized`

Reply:
Moved to syncCommit logic and added javadoc there.